### PR TITLE
Feature: extract the feature flag for option-based form editor from campaigns domain into the core

### PR DESCRIPTION
**Pull request by glaubersilva**
Resolves [GIVE-1345]

## Description

This PR extracts from [the campaigns epic branch](https://github.com/impress-org/givewp/pull/7506) the feature to enable or disable the Option-Based Form editor. The main goal here is to be able to release this new feature as soon as possible, without the need to wait until the Campaigns project is finished.

This PR adds a new option to the `Give > Settings > Advanced` page that allows users to enable or disable the Option-Based Form Editor. This new option is enabled by default if the website has any Option-Based forms on DB and disabled by default if it has only Visual Form Builder forms on DB.

## Affects

The "Settings" page and the "Switch to Legacy View" button from the "All Forms" page.

## Visuals

**Demo video:**

https://www.loom.com/share/6b1063475387458a9621a24c920b8077?sid=914ffc3f-19a2-4159-be82-ed2868e90eca

## Testing Instructions

Go to the `Give > Settings > Advanced` page to enable or disable the `Option-Based Form Editor` option and make sure the expected behaviors occur.

**Expected behaviors with the legacy editor disabled:** 

- Options used only by the legacy editor are hidden.
- The "Switch to Legacy View" button is hidden on the "All Forms" page.

**Expected behaviors with the legacy editor enabled:** 

- Options used only by the legacy editor have a helper text icon next to the label.

## Pre-review Checklist

-   [ ] Acceptance criteria satisfied and marked in related issue
-   [ ] Relevant `@unreleased` tags included in DocBlocks
-   [ ] Includes unit tests
-   [ ] Reviewed by the designer (if follows a design)
-   [ ] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

[GIVE-1345]: https://stellarwp.atlassian.net/browse/GIVE-1345?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by glaubersilva** [comment]
@jonwaldstein Gentle ping here to remind you about the double-check that we talked another day related to the settings options audit that should be visible only when the Option-based form editor is enabled. So, when you have the chance please let me know if we have hidden all necessary v2 options or if something was left behind.

File: src/FeatureFlags/OptionBasedFormEditor/Settings/General.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+{
+    /**
+     * @unreleased
+     */
+    public function getDisabledOptionIds(): array
+    {
+        return [
+            //General Section
+            'success_page',
+            'failure_page',
```
Comment (jonwaldstein) [inline src/FeatureFlags/OptionBasedFormEditor/Settings/General.php:18]: We can keep these (recently added support for success page)

File: src/FeatureFlags/OptionBasedFormEditor/Settings/General.php [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        ];
+    }
+
+    /**
+     * @unreleased
+     */
+    public function getDisabledSectionIds(): array
+    {
+        return [
+            'access-control',
```
Comment (jonwaldstein) [inline src/FeatureFlags/OptionBasedFormEditor/Settings/General.php:35]: So I think we actually need to double check this section.  I know I originally thought these were legacy but now I'm pretty sure some of the settings here are actually used in the donor dashboard like the "Email access" setting.

File: src/FeatureFlags/OptionBasedFormEditor/Settings/DefaultOptions.php [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        ];
+    }
+
+    /**
+     * @unreleased
+     */
+    public function getDisabledSectionIds(): array
+    {
+        return [
+            'display-settings',
```
Comment (jonwaldstein) [inline src/FeatureFlags/OptionBasedFormEditor/Settings/DefaultOptions.php:31]: This one turns out we actually do use one setting 😬 :
- `Title Prefixes`: currently used in the donor name block

File: src/FeatureFlags/OptionBasedFormEditor/Settings/DefaultOptions.php [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+            'taxonomies',
```
Comment (jonwaldstein) [inline src/FeatureFlags/OptionBasedFormEditor/Settings/DefaultOptions.php:32]: We can keep this, we now use this taxonomies section in v3 forms

File: src/FeatureFlags/OptionBasedFormEditor/Settings/General.php [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+            //General Section
+            'success_page',
+            'failure_page',
+            // Currency Section
+            'auto_format_currency',
+            'currency_position',
+            'thousands_separator',
+            'decimal_separator',
+            'number_decimals',
+            'currency_preview',
```
Comment (jonwaldstein) [inline src/FeatureFlags/OptionBasedFormEditor/Settings/General.php:25]: I think we can add `override_legacy_donation_management_pages` here as well.  I'm going to ask about the donation history shortcode https://givewp.com/documentation/core/shortcodes/donation_history/

**Review by jonwaldstein — CHANGES_REQUESTED** [review]
@glaubersilva this is great! I noted some changes to settings that have changed since my last audit 😄

File: src/FeatureFlags/OptionBasedFormEditor/Settings/General.php (hunk 1 above)
Comment (glaubersilva) [reply src/FeatureFlags/OptionBasedFormEditor/Settings/General.php]: Resolved in 512ad4dc6a0e8c13cfb06b4553ddf969ab81e0fd

File: src/FeatureFlags/OptionBasedFormEditor/Settings/DefaultOptions.php (hunk 3 above)
Comment (glaubersilva) [reply src/FeatureFlags/OptionBasedFormEditor/Settings/DefaultOptions.php]: Resolved in 512ad4dc6a0e8c13cfb06b4553ddf969ab81e0fd

File: src/FeatureFlags/OptionBasedFormEditor/Settings/General.php (hunk 5 above)
Comment (glaubersilva) [reply src/FeatureFlags/OptionBasedFormEditor/Settings/General.php:24]: Resolved in 512ad4dc6a0e8c13cfb06b4553ddf969ab81e0fd

File: src/FeatureFlags/OptionBasedFormEditor/Settings/General.php (hunk 2 above)
Comment (glaubersilva) [reply src/FeatureFlags/OptionBasedFormEditor/Settings/General.php]: Resolved in af1cd458b17269d5f81a62c0c062e0059823cf1f and 7c3467d3a59c972e76f51f669671ccc8b48cf18c

**Comment by glaubersilva** [comment]
@jonwaldstein I implemented the changes based on your last audit. Let me know if there is something else.

File: assets/src/css/admin/settings.scss [hunk 6]
```diff
@@ -383,6 +383,90 @@ div.give-field-description {
...
+}
+
+.give-setting-tab-body-general,
+.give-setting-tab-body-display,
+.give-settings-advanced-tab {
+	label {
+		display: flex;
+		gap: 0.4rem;
+		padding: 1.25rem 1rem;
+		position: relative;
```
[the remark this answers is not in the document]
Comment (glaubersilva) [reply assets/src/css/admin/settings.scss]: Resolved in 028a9b1a569ae6d89ae86a35b1f1fb3f43b26060

**Review by jonwaldstein — APPROVED** [review]
Great job! @glaubersilva

**Review by rickalday — APPROVED** [review]
Passed manual QA tests.

**Pull request merged**